Src/Newtonsoft.Json/Converters/XmlNodeConverter.cs

```diff
@@ -1177,6 +1177,11 @@ private bool IsArray(IXmlNode node)
 
         private void SerializeGroupedNodes(JsonWriter writer, IXmlNode node, XmlNamespaceManager manager, bool writePropertyName)
         {
+            if (node.ChildNodes.Count == 0)
```

Comment:
Yes, I tried doing a `HasAttributes` but as well as it being guarded against for the null case already it doesn't offer as much of an improvement. The `HasChildNodes` test is in now, though.